### PR TITLE
Adding Styled-Components

### PR DESCRIPTION
**Pull request by iamalisajid**
Adding styled components to the whole application along with the theme switch.
Modules Done: All
Adding storybook
Modules Done: Todos
**Demo:** 
APP: https://www.useloom.com/share/d9e2c6ceea444d28a8b105c621fc43d3
Story: https://www.useloom.com/share/9d2eab744d34475992bd62f47a481664

### CODE REVIEW

File: .storybook/config.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import { configure } from '@storybook/react';
+
+function loadStories() {
+  require('../src/stories');
```
Comment (InamTaj) [inline .storybook/config.js]: Shouldn't be from a specific directory. Pick any file that matches the regex `*.story.js`

File: src/globalStyles/index.js [hunk 2]
```diff
@@ -0,0 +1,182 @@
+import styled from 'styled-components';
+import { Link } from 'react-router-dom';
+import { createGlobalStyle } from 'styled-components';
+
+export const GlobalStyle = createGlobalStyle`
+  body {
+    background-image: url('/bg.png');
+    font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", "Roboto", "Oxygen",
```
Comment (InamTaj) [inline src/globalStyles/index.js]: Use only 2 fonts as `primary-font` and secondary font. Remove browser specific CSS styling.

File: src/globalStyles/index.js [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+  }
+  font-size: 1rem;
+  font-weight: 400;
+  line-height: 1.5;
+  color: #212529;
+  text-align: left;
+`;
+
+export const Button = styled.button`
+  display: inline-block;
```
Comment (InamTaj) [inline src/globalStyles/index.js]: Sort alphabetically. Integrate linting for styled components.

File: src/reducers/todos.js [hunk 4]
```diff
@@ -30,7 +30,7 @@ const todos = (state = initialState.todos, action) => {
       return {
         ...state,
         loading: false,
-        todos: state.todos.map(todo => todo.id === action.payload.id ? action.payload : todo),
+        todos: state.todos.map(todo => todo.id === action.payload.id ? action.payload : todo)
```
Comment (InamTaj) [inline src/reducers/todos.js]: Revert this and fix ESLint config to allow trailing commas.

File: src/stories/TodoList/index.js [hunk 5]
```diff
@@ -0,0 +1,19 @@
+import React from 'react';
```
Comment (InamTaj) [inline src/stories/TodoList/index.js]: rename `src/stories` to `src/__stories__`

File: src/stories/index.js [hunk 6]
```diff
@@ -0,0 +1,4 @@
+export * from './TodoItem';
```
Comment (InamTaj) [inline src/stories/index.js]: No need to have this file.

File: src/viewes/Contacts/styles.js [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+export const FirstName = styled.div`
+  grid-column: span 6;
+`;
+export const LastName = styled.div`
+  grid-column: span 6;
+`;
+export const ContactImage = styled.div`
+  grid-column: span 4;
+`;
+export const ContactDetails = styled.div`
```
Comment (InamTaj) [inline src/viewes/Contacts/styles.js]: Create a generic component of div with flexible grid-column property.

File: src/viewes/Dashboard/index.js [hunk 8]
```diff
@@ -1,27 +1,18 @@
...
-      </div>
-    </div>
-  </div>
+  <Fragment>
+    <Welcome>
+      <h1>{WELCOME.TITLE}</h1>
+      {WELCOME.MESSAGE}
+      <ButtonGroup>
+        <NavButton success as="a" href="/todos">Todos</NavButton>
+        <NavButton primary as="a" href="/contacts">Contacts</NavButton>
```
Comment (InamTaj) [inline src/viewes/Dashboard/index.js]: Change URLs to constants.

File: src/viewes/Login/LoginForm/index.js [hunk 9]
```diff
@@ -1,41 +1,45 @@
...
-          <div className="custom-control custom-checkbox mb-3">
-            <input type="checkbox" className="custom-control-input" id="customCheck1"/>
-            <label className="custom-control-label" htmlFor="customCheck1">Remember password</label>
-          </div>
-          <button className="btn btn-lg btn-primary btn-block text-uppercase" type="submit">Sign in</button>
-        </form>
-      </div>
-    </div>
+          </FormLabelGroup>
+          <Error> {error} </Error>
```
Comment (InamTaj) [inline src/viewes/Login/LoginForm/index.js]: No surrounding spaces

File: src/viewes/Login/styles.js [hunk 10]
```diff
@@ -0,0 +1,104 @@
+import styled from 'styled-components';
+import { Button, Card } from '../../globalStyles';
+
+export const LoginFormLayout = styled.div`
+  grid-column: 4 / span 6;
+  margin-left: auto!important;
+  margin-right: auto!important;
+  --input-padding-x: 1.5rem;
+  --input-padding-y: .75rem;
```
Comment (InamTaj) [inline src/viewes/Login/styles.js]: Please remove these two styles and use standard CSS

File: src/viewes/Login/styles.js [hunk 11]
```diff
@@ -0,0 +1,104 @@
...
+  width: 100%;
+  display: block;
+  border-radius: 5rem;
+  font-size: 80%;
+  font-weight: bold;
+  letter-spacing: .1rem;
+  padding: 1rem;
+  transition: all 0.2s;
+  margin-top: 10%;
+  text-transform: uppercase!important;
```
Comment (InamTaj) [inline src/viewes/Login/styles.js]: sort and lint

File: src/viewes/Login/styles.js (hunk 10 above)
Comment (iamalisajid) [reply src/viewes/Login/styles.js]: actually they are used in several places for further calculations.

**Pull request merged**